kolibri/plugins/document_pdf_render/assets/src/views/index.vue

```diff
@@ -177,28 +176,17 @@
             index: i,
           });
         }
-
         return this.getPage(1).then(firstPage => {
-          const pageMargin = 8;
-          this.firstPageWidth = firstPage.view[2];
           this.firstPageHeight = firstPage.view[3];
-
-          this.scale =
-            this.windowSize.breakpoint > 3
-              ? 1
-              : (this.elSize.width - 2 * pageMargin) / this.firstPageWidth;
-
-          // set default height and width properties
-          const initialViewport = firstPage.getViewport(this.scale);
-          this.pageHeight = initialViewport.height;
-          this.pageWidth = initialViewport.width;
+          this.firstPageWidth = firstPage.view[2];
+          this.scale = this.elSize.height / (this.firstPageHeight + MARGIN);
```

Comment:
Ah, so we dynamically set the scale, based on the size of the page and the available viewport, rather than setting the viewport by the scale? Neat.

Reply:
Yup, it isn't ideal, but it is definitely much better than loading the entire pdf. Plus we have those handy zoom buttons.